aicsimageio/readers/ome_zarr_reader.py

```diff
@@ -0,0 +1,144 @@
+    fs_kwargs: Dict[str, Any]
+        Any specific keyword arguments to pass down to the fsspec created filesystem.
+        Default: {}
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            ZarrReader(parse_url(path, mode="r"))
+            return True
+
+        except AttributeError:
+            return False
+
+    def __init__(
+        self,
+        image: types.PathLike,
+        fs_kwargs: Dict[str, Any] = {},
+    ):
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(
+            image,
+            enforce_exists=False,
+            fs_kwargs=fs_kwargs,
+        )
+
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+        self._zarr = ZarrReader(parse_url(self._path, mode="r")).zarr
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            scenes = self._zarr.root_attrs["multiscales"]
+
+            # if (each scene has a name) and (that name is unique) use name.
+            # otherwise generate scene names.
+            if all("name" in scene for scene in scenes) and (
+                len({scene["name"] for scene in scenes}) == len(scenes)
+            ):
+                self._scenes = tuple(str(scene["name"]) for scene in scenes)
+            else:
+                self._scenes = tuple(
+                    metadata_utils.generate_ome_image_id(i)
+                    for i in range(len(self._zarr.root_attrs["multiscales"]))
+                )
+        return self._scenes
+
+    def _read_delayed(self) -> xr.DataArray:
+        return self._xarr_format(delayed=True)
+
+    def _read_immediate(self) -> xr.DataArray:
+        return self._xarr_format(delayed=False)
+
+    def _xarr_format(self, delayed: bool) -> xr.DataArray:
+        image_data = self._zarr.load(str(self.current_scene_index))
+        axes = self._zarr.root_attrs["multiscales"][self.current_scene_index]["axes"]
+        dims = [sub["name"].upper() for sub in axes]
+
+        if not delayed:
+            image_data = image_data.compute()
+
+        coords = self._get_coords(
+            dims,
+            image_data.shape,
+            scene=self.current_scene,
+            channel_names=self._get_channel_names_from_ome(),
+        )
+
+        return xr.DataArray(
+            image_data,
+            dims=dims,
+            coords=coords,
+            attrs={constants.METADATA_UNPROCESSED: self._zarr.root_attrs},
+        )
+
+    @staticmethod
+    def _get_coords(
+        dims: List[str],
+        shape: Tuple[int, ...],
+        scene: str,
+        channel_names: Optional[List[str]],
+    ) -> Dict[str, Any]:
+        # Use dims for coord determination
+        coords: Dict[str, Any] = {}
```

Comment:
is there some missing implementation in this function?

Reply:
Coords are just the xarray coords. I think this is fine for now? The other coord we commonly add is timeseries info.

Reply:
I see, I thought there was more work to do in here to incorporate physical pixel dims or something.